cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -314,9 +315,8 @@ type JoinConfiguration struct {
 	// control plane instance.
 	ControlPlane bool
 
-	// AdvertiseAddress sets the IP address for the API server to advertise; the
-	// API server will be installed only on nodes hosting an additional control plane instance.
-	AdvertiseAddress string
+	// APIEndpoint represents the endpoint of the instance of the API server eventually to be deployed on this node.
+	APIEndpoint APIEndpoint
```

Comment:
The Naming is confusing here. 

The APIEndpoint struct has an APIEndpoint member.

Reply:
@timothysc  Sorry but I don't understand you comment :confused:

APIEndpoint struct has two attributes, AdvertiseAddress and BindPort:

Here we have an attribute that has the same name of its type. is that the thing you find confusing? Do you prefer naming the struct `APIEndpointConfiguration`

Reply:
Sorry, the way github collapsed this changeset made me misinterpret this.  I expanded it looks ok.

Reply:
Or... are you looking for the "status" we discusse at the end of the call yesterday?

If this is the case, please consider that this PR implements only the "spec", while the status will be implemented the next PR (in the form of `map[bindaddress]APIEndoint`, but I'm figuring out details right now)

Reply:
> map[bindaddress]APIEndoint

i though more about this. we ideally need to map against the address the user provides instead of always defaulting to the address of the network interface, because we do have `--apiserver-advertise-address`.

if i understand this correctly this might result in an interesting situation, but i guess we can comment more in the next PR. :\